### PR TITLE
[cli]: Add package_metadata.rs:

### PR DESCRIPTION
**Pull request by itayadler**
- package_metadata.rs returns a PackageMetadata struct
that represents the Buff toml file. It exposes
`load_from_path` to load a buff file and returns
the struct.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #[macro_use]
 extern crate clap;
+#[macro_use]
```
Comment (BlueHotDog) [inline cli/src/main.rs]: this means you import the macros defined in this crate?

File: cli/src/package_metadata.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use std::fs;
+
+#[derive(Deserialize)]
+pub struct PackageMetadata {
+  package: Package,
+  dependencies: HashMap<String, String>,
+}
+
+#[derive(Deserialize)]
+struct Package {
```
Comment (BlueHotDog) [inline cli/src/package_metadata.rs]: why the split?

File: cli/src/main.rs (hunk 1 above)
Comment (itayadler) [reply cli/src/main.rs]: I read about macros and it seems like its no longer needed if we use rust 2018, will fix

File: cli/src/package_metadata.rs (hunk 2 above)
Comment (itayadler) [reply cli/src/package_metadata.rs]: the main struct holds both the package section and the dependencies section, if you have an idea for a better name shoot

**Pull request closed without merging**